x/crosschain/keeper/msg_server_add_to_intx_tracker_test.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/stretchr/testify/require"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/common/ethereum"
+	keepertest "github.com/zeta-chain/zetacore/testutil/keeper"
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/x/crosschain/keeper"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observerTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestMsgServer_AddToInTxTracker(t *testing.T) {
+	t.Run("add proof based tracker with correct proof", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		chainID := int64(5)
+		txIndex, block, header, headerRLP, proof, tx, err := sample.Proof()
+		require.NoError(t, err)
+		setupVerificationParams(zk, ctx, txIndex, chainID, header, headerRLP, block)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err = msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   sample.AccAddress(),
+			ChainId:   chainID,
+			TxHash:    tx.Hash().Hex(),
+			CoinType:  common.CoinType_Zeta,
+			Proof:     proof,
+			BlockHash: block.Hash().Hex(),
+			TxIndex:   txIndex,
+		})
+		require.NoError(t, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx.Hash().Hex())
+		require.True(t, found)
+	})
+
+	t.Run("fail to add proof based tracker with wrong tx hash", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		chainID := int64(5)
+		txIndex, block, header, headerRLP, proof, tx, err := sample.Proof()
+		require.NoError(t, err)
+		setupVerificationParams(zk, ctx, txIndex, chainID, header, headerRLP, block)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err = msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   sample.AccAddress(),
+			ChainId:   chainID,
+			TxHash:    "fake_hash",
+			CoinType:  common.CoinType_Zeta,
+			Proof:     proof,
+			BlockHash: block.Hash().Hex(),
+			TxIndex:   txIndex,
+		})
+		require.ErrorIs(t, types.ErrTxBodyVerificationFail, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx.Hash().Hex())
+		require.False(t, found)
+	})
+
+	t.Run("fail to add proof based tracker with wrong chain id", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		chainID := int64(5)
+		txIndex, block, header, headerRLP, proof, tx, err := sample.Proof()
+		require.NoError(t, err)
+		setupVerificationParams(zk, ctx, txIndex, chainID, header, headerRLP, block)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		_, err = msgServer.AddToInTxTracker(ctx, &types.MsgAddToInTxTracker{
+			Creator:   sample.AccAddress(),
+			ChainId:   97,
+			TxHash:    tx.Hash().Hex(),
+			CoinType:  common.CoinType_Zeta,
+			Proof:     proof,
+			BlockHash: block.Hash().Hex(),
+			TxIndex:   txIndex,
+		})
+		require.ErrorIs(t, types.ErrTxBodyVerificationFail, err)
+		_, found := k.GetInTxTracker(ctx, chainID, tx.Hash().Hex())
+		require.False(t, found)
+	})
+
+	t.Run("fail to add proof based tracker with wrong proof", func(t *testing.T) {
```

Comment:
which part of the proof was faked out (or set to wrong)

Reply:
the proof is NewProof() , not the one generated from txid